torch/_dynamo/output_graph.py

```diff
@@ -1015,3 +1060,51 @@ def create_graph_input(self, name, type_expr=None, before=False):
             else:
                 self.input_name_to_proxy[name] = proxy
             return proxy
+
+    def is_name_bound(self, name):
+        if name in self.input_name_to_proxy:
+            return True
+        for proxy in self.seen_proxies:
+            if proxy.node.name == name:
+                return True
+        return False
+
+    def lift_freevar_to_input(self, proxy):
```

Comment:
As noted above - you only call this under `if self.parent is not None:` 

Either make the invariant that this is safe to call at any time (Like you have on L1075) or assert that self.parent is not None here and call it unequivocally after name bound checking.